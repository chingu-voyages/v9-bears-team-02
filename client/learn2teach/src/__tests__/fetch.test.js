import React from 'react';
import { render, cleanup,fireEvent,waitForElement } from '@testing-library/react';
import 'jest-dom/extend-expect';
import Fetch from '../utils/fetch';
afterEach(cleanup)
test('Fetch makes an API call and displays the greeting when load-greeting is clicked', async ()=>{
    //Arrange
    const axiosMock = jest
    .fn()
    .mockResolvedValueOnce({data: {greeting: 'hello there'}})
    const url = '/greeting'
    const { getByText, getByTestId, container, asFragment } = render(<Fetch url={url} />)
    //Act
    fireEvent.click(getByText('Load Greeting'))
    const greetingTextNode = await waitForElement(()=>
       getByTestId('greeting-text'));
    //Assert
    //expect(axiosMock.get).toHaveBeenCalledTimes(1)
    //expect(axiosMock.get).toHaveBeenCalledWith(url)
    expect(getByTestId('greeting-text')).toHaveTextContent('hello there')
    expect(getByTestId('ok-button')).toHaveAttribute('disabled')

    // snapshots work great with regular DOM nodes!
    expect(container.firstChild).toMatchSnapshot()

    // you can also use get a `DocumentFragment`,
    // which is useful if you want to compare nodes across render
   expect(asFragment()).toMatchSnapshot()
})


