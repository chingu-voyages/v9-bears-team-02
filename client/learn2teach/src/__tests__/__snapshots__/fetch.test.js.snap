// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch makes an API call and displays the greeting when load-greeting is clicked 1`] = `
<div
  data-testid="greeting-text"
>
  <h1>
    Load Greeting
  </h1>
  <p>
    hello there
  </p>
  <button
    data-testid="ok-button"
    disabled=""
  >
    Test
  </button>
</div>
`;

exports[`Fetch makes an API call and displays the greeting when load-greeting is clicked 2`] = `
<DocumentFragment>
  <div
    data-testid="greeting-text"
  >
    <h1>
      Load Greeting
    </h1>
    <p>
      hello there
    </p>
    <button
      data-testid="ok-button"
      disabled=""
    >
      Test
    </button>
  </div>
</DocumentFragment>
`;
